server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -414,11 +420,6 @@ static void adjustSearchType(SearchRequest searchRequest, boolean singleShard) {
             searchRequest.requestCache(false);
             searchRequest.searchType(QUERY_THEN_FETCH);
         }
-
```

Comment:
should we also move up the isSuggestOnly branch and return from it?

Reply:
Yes! That also makes sense. I will make the change.